qutebrowser/browser/webkit/network/networkmanager.py

```diff
@@ -270,28 +268,11 @@ def clear_rejected_ssl_errors(self, url):
     @pyqtSlot('QNetworkReply*', 'QAuthenticator*')
     def on_authentication_required(self, reply, authenticator):
         """Called when a website needs authentication."""
-        user, password = None, None
-        if not hasattr(reply, "netrc_used") and 'HOME' in os.environ:
-            # We'll get an OSError by netrc if 'HOME' isn't available in
-            # os.environ. We don't want to log that, so we prevent it
-            # altogether.
-            reply.netrc_used = True
-            try:
-                net = netrc.netrc(config.val.content.netrc_file)
-                authenticators = net.authenticators(reply.url().host())
-                if authenticators is not None:
-                    (user, _account, password) = authenticators
-            except FileNotFoundError:
-                log.misc.debug("No .netrc file found")
-            except OSError:
-                log.misc.exception("Unable to read the netrc file")
-            except netrc.NetrcParseError:
-                log.misc.exception("Error when parsing the netrc file")
-
-        if user is not None:
-            authenticator.setUser(user)
-            authenticator.setPassword(password)
-        else:
+        netrc = False
+        if not hasattr(self, 'netrc_used'):
+            setattr(self, 'netrc_used', True)
+            netrc = shared.netrc_authentication(reply.url(), authenticator)
```

Comment:
The comments from above about `netrc` and `settatr`/`hasattr` apply here as well. Also, it should be a private attribute here (`self._netrc_used`) as it's not used from the outside.

Reply:
It's actually accessed from the webkittab class in its _on_load_started() method, so I cannot make this private.